src/Interpreters/ActionsDAG.cpp

```diff
@@ -1837,10 +1828,8 @@ ActionsDAG::SplitResult ActionsDAG::split(std::unordered_set<const Node *> split
     for (const auto * input : new_inputs)
     {
         const auto & cur = data[input];
-        if (cur.to_second)
-            second_inputs.push_back(cur.to_second);
-        if (cur.to_first)
-            first_outputs.push_back(cur.to_first);
+        second_inputs.push_back(cur.to_second);
+        first_outputs.push_back(cur.to_first);
```

Comment:
Wouldn't this change the result of this function? As far as I understood some if these pointers might be nullptr, which sounds bad for me.

Reply:
Ideally, this should not happen. Or it would immediately cause a crash :)
But they should not be nullptr; it's a kind of invariant I expect here.
Basically, new inputs are those nodes that were not in the list of inputs initially (for splitted DAG), butappear as intermediate result. In this case, to_first would point to some node of the first DAG, and to_second would point to some input of the second DAG.

Reply:
Good point @Algunenano, thanks!